src/ty/variant.rs

```diff
@@ -222,4 +225,9 @@ where
     pub fn discriminant(&self) -> Option<u64> {
         self.discriminant
     }
+
+    /// Returns the documentation of the variant.
+    pub fn docs(&self) -> &[T::String] {
+        &self.docs
```

Comment:
Maybe `doc` should be an `Option`. Feels a bit more rusty to check for `None` rather than an empty slice?

Reply:
I prefer an empty slice, it conveys the same meaning and is easier to consume.